src/framework/preferences/js/LocalizationEnactor.js

```diff
@@ -0,0 +1,152 @@
+        gradeNames: ["fluid.prefs.enactor", "fluid.contextAware", "fluid.resolveRoot"],
+        preferenceMap: {
+            "fluid.prefs.localization": {
+                "model.lang": "value"
+            }
+        },
+        contextAwareness: {
+            localeChange: {
+                checks: {
+                    inPanel: {
+                        contextValue: "{iframeRenderer}"
+                    },
+                    urlPath: {
+                        contextValue: "{localization}.options.localizationScheme",
+                        equals: "urlPath",
+                        gradeNames: "fluid.prefs.enactor.localization.urlPathLocale",
+                        priority: "after:inPanel"
+                    }
+                },
+                defaultGradeNames: "fluid.prefs.enactor.localization.no-op"
+            }
+        }
+    });
+
+    /*******************************************************************************
+     * URL Path
+     *
+     * Changes the URL path to specify which language should be displayed. Useful
+     * if languages are served at different URLs based on a language resource.
+     * E.g. www.example.com/about/ -> www.example.com/fr/about/
+     *******************************************************************************/
+    fluid.defaults("fluid.prefs.enactor.localization.urlPathLocale", {
+        langMap: {}, // must be supplied by integrator
+        langSegIndex: 1,
+        model: {
+            urlPathname: undefined
+        },
+        listeners: {
+            "onCreate.setURLPath": {
+                changePath: "urlPathname",
+                value: {
+                    expander: {
+                        func: "{that}.pathname"
+                    }
+                },
+                source: "initialURLPathname"
+            }
+        },
+        modelListeners: {
+            lang: {
+                funcName: "fluid.prefs.enactor.localization.urlPathLocale.changeLocale",
+                args: ["{that}", "{change}.value", "{that}.options.langMap", "{that}.options.langSegIndex"],
+                namespace: "langToURLPath"
+            },
+            urlPathname: {
+                funcName: "{that}.pathname",
+                args: ["{change}.value"],
+                excludeSource: ["init", "initialURLPathname"],
+                namespace: "setURLPathname"
+            }
+        },
+        invokers: {
+            pathname: "fluid.prefs.enactor.localization.urlPathLocale.pathname"
+        }
+    });
+
+    /**
+     * A simple wrapper around the location.pathname setter/getter property.
+     *
+     * @param {String} pathname - (Optional) pathname to set. If not supplied the current pathname will be returned
+     * @return {String} - If the `pathname` argument is not provided, the current pathname is returned
+     */
+    fluid.prefs.enactor.localization.urlPathLocale.pathname = function (pathname) {
```

Comment:
It might be clearer to define setPathName() and getPathName() rather than sharing these 2 functionalities via this single function. It seems the getter is used for setting the initial value at line 70 and the setter is for responding to path change requests at line 83.